### DISCUSSION TITLE
Common hardware for standalone debugger

### DISCUSSION THREAD

**Discussion by FlorianUekermann** [ARMmbed/DAPLink#964] [General]
Has DAPLink been ported to any very common boards or debuggers for use as a standalone debugger? And are those builds included in the releases? If so, a hint in the readme would go a long way to point people in the right direction.

I'm thinking about something like the stm32f103c8t6 based blue pill or the official rp pico board. Many people have stuff like that lying around and those boards are much easier to source for hobbyists from Amazon or the electronics store around the corner, compared to the options I have found so far.
I have seen that there are some forks claiming compatibility, but couldn't find any indication of upstreaming efforts or sufficient documentation. Support in the DAPLink repo would also be a lot less sketchy and suggest a higher probability of long-term maintenance.

**Comment by mathias-arm** [comment]
- Some STM32F103CBxx boards (like STLinkV2.1 in Nucleo board, or clones) should be supported by `stm32f103xb_bl` and `stm32f103xb_if`.
- I have been thinking about a STM32F103C8T6 variant of `stm32f103xb_if` (without bootloader) that would work with boards like "Blue Pill".
- I have be hacking on support for WeAct Studio "Black Pill" boards (https://github.com/mbrossard/DAPLink/tree/feature/stm32f4x1ce_hic but not ready for prime time). The MCU is limited by the number of endpoints, so it's not possible to have all features at the same time.
- Raspberry Pi Pico support would be very nice. The main hurdle would be the USB driver, we can't just use the tinyUSB one. If someone is willing to take on this project, I will gladly assist.

**Reply by raveslave** [reply]
The Zephyr RP2040 port has USB support. An option perhaps?

**Reply by FlorianUekermann** [reply]
I have moved on to https://github.com/majbthrd/DapperMime, which works great for my purposes on the Pico. Maybe you can get a few pointers from there.

Given the strong support from ARM for this project implied by the README, I'm a bit surprised by the lack of interest to support or document at least one cheap and popular board to lower the barrier of entry to the ARM ecosystem. There also hasn't been a commit since I opened the issue... If this project has been abandoned, maybe someone should put a note in the README.
